src/renderer/components/editorWithTabs/editor.vue

```diff
@@ -158,7 +158,7 @@ export default {
       imageInsertAction: state => state.preferences.imageInsertAction,
       imagePreferRelativeDirectory: state => state.preferences.imagePreferRelativeDirectory,
       imageRelativeDirectoryName: state => state.preferences.imageRelativeDirectoryName,
-      imageFolderPath: state => state.preferences.imageFolderPath,
+      imageFolderPath: state => state.preferences.imageFolderPath.replace(/\${filename}/g, this.currentFile),
```

Comment:
Likely you want just the filename `this.currentFile.filename` but this would result in an exception at runtime because `this.currentFile` is not available due to the context. You have to use `state.editor.currentFile.filename`. Better would be a utility function (in `src/renderer/util`) that replaces a given path with the given filename. This would outsource the code that's required in other locations as well and makes it easier to understand what is done here.

@Jocs Further, I think we shouldn't replace the `imageFolderPath` variable because it's a settings entry. I would prefer adding an additional one or better to replace it inside `editor.vue@imageAction` and search other locations as well. `imageAction` is the core module that consumes the both absolute and relative paths.